src/java.base/share/classes/jdk/internal/crac/JDKResource.java

```diff
@@ -29,5 +29,23 @@
 import jdk.crac.Resource;
 
 public interface JDKResource extends Resource {
-    int getPriority();
+    /**
+     * JDK Resource priorities.
+     * Most resources should use NORMAL.
+     * Other priorities define sequence of
+     * checkpoint notification for dependent resources
+     */
+    enum Priority {
+        /**
+         * Priority of the
+         * sun.nio.ch.EPollSelectorImpl resource
+         */
+        EPOLLSELECTOR,
```

Comment:
This uncovers the future mess we'll have with manually setting the order, similar to the mutex order in the Hotspot. However, I like this central place to control them. This approach is by no means worse than the existing code but highlights a point of growth.